app/api/views.py

```diff
@@ -150,14 +150,40 @@ def getspaces():
 @login_required
 def addproducts(current_user):
     data = request.get_json()
-    new_product = Product(name=data['productName'],description=data['productDesc'],sale_unit=data['sale_unit'],price=data['price'],images=data['images'],Instock=data['available_stock'],discount=data['discount'],date_created=d.datetime.utcnow())
+    new_product = Product(name=data['productName'],description=data['productDesc'],sale_unit=data['sale_unit'],price=data['price'],images=data['images'],Instock=data['available_stock'],discount=data['discount'],date_created=d.datetime.utcnow(), productID=str(uuid4()))
     db.session.add(new_product)
     # fetch store 
     current_space = current_user.space
     new_product.space = current_space
     db.session.commit()
     return jsonify({'msg':f'Added {new_product.name} successfully'}),200
 
+@api.route('/getallproducts', methods=['GET'])
+@login_required
+def getallproducts(current_user):
```

Comment:
This getallproducts is explicit to every product in our database and should be accessible by an admin only, otherwise ensure that you're doing this for a specific space : this would mean that you need to add a second argument to streamline your query. Perhaps you might want to do this on a new route such as : "/getproducts/<storeId>"
```suggestion
def getallproducts(current_user):
        if current_user.email == current_app.config['FLASK_ADMIN'] or current_user.role.has_permission(PERMISSION.ADMIN):
```

Reply:
Alright. I'll add the functionality for an admin and a user (this would return all products belonging to a user in the database and a separate route for all products belonging to particular space for a user)